utils/polus-python-template/hooks/post_gen_project.py

```diff
@@ -0,0 +1,76 @@
+import os
+import shutil
+from pathlib import Path
+import logging
+from os import environ
+
+logging.basicConfig(
+    format="%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s",
+    datefmt="%d-%b-%y %H:%M:%S",
+)
+POLUS_LOG = getattr(logging, environ.get("POLUS_LOG", "DEBUG"))
+logger = logging.getLogger("polus-python-template-post")
+logger.setLevel(POLUS_LOG)
+
+"""Create the final plugin directory structure.
+(ex: polus/plugins/package1/package2/awesome_function)
+and copy our generated plugin there.
+"""
+# rename the original source directory
+os.rename("src", "src_template")
+# create the new source directory
+package_dir = "{{ cookiecutter.plugin_package }}"
+package_dir = package_dir.split(".")
+package_dir = os.path.join(*package_dir)
+
+src = Path("src") / package_dir
+os.makedirs(src, exist_ok=False)
+# copy old source in new source structure
+src_template = Path("src_template")
+files = os.listdir(src_template)
+for file_name in files:
+    shutil.move(src_template / file_name, src)
+# remove the original source directory
+os.rmdir(src_template)
+
+logger.debug(f"created source directories : {src}")
+
+""" Buid the correct directories inside polus-plugins.
+The directory structure must conforms to the plugin's spec :
+    - dash-separated word in identifier.
+    - folder hierarchy matches package namespace minus "polus.plugins"
+    - plugin's folder name reflects the plugin package name but ends with "-plugin"
+Ex: polus.plugins.package1.package2.awesome_function becomes
+package1/package2/awesome-function-plugin
+"""
+# keep track of our new project
+source_dir = Path(os.getcwd())
+
+# try to find the project's root, otherwise we stay in the
+# staging directory
+final_dir = source_dir.parent
+for folder in Path(final_dir).parent.parents:
+    if os.path.exists(folder / ".git"):
+        final_dir = folder
+        break
```

Comment:
I think this check is fragile, if we ever have nested repos or add git submodules. I think we should check the `.git/config` file for a ref to a fork of the `polus-plugins` repo. That would still not be infallible but should be a lot better than this.